resources/devbox/macos/macos.sh

```diff
@@ -0,0 +1,149 @@
+            ;;
+        kmcomp)
+            $REQUIRE_KMCOMP=true
+            ;;
+        pandoc)
+            $REQUIRE_PANDOC=true
+            ;;
+        sentry-cli)
+            $REQUIRE_SENTRYCLI=true
+            ;;
+        all)
+            REQUIRE_ANDROID=true
+            REQUIRE_IOS=true
+            REQUIRE_MACOS=true
+            REQUIRE_WEB=true
+            ;;
+        *)
+            echo "Error: unrecognised parameter. Use --help for help"
+            ;;
+    esac
+    shift
+done
+
+# This script will configure your development environment from a bare metal mac install. It should be idempotent.
+
+echo "This script will configure your macOS computer to build Keyman, installing build tools and prerequisites."
+echo "You can also do this yourself following the notes in building.md."
+read -p "Press ENTER to start install"
+
+if $REQUIRE_IOS || $REQUIRE_ANDROID; then
+  REQUIRE_WEB=true
+fi
+
+if $REQUIRE_ANDROID; then
+  REQUIRE_PANDOC=true
+fi
+
+## These components are required by all development toolchains
+
+# XCode command lines tools will be installed by Brew
+# xcode-select --install
+
+which brew || (
+    # Install Homebrew
+    /bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh)"
+) && (
+    brew update
+)
+
+## Install devchain components
+
+BREW_ALL="bash jq python3 meson ninja rustup-init coreutils"
+BREW_WEB="node emscripten wasm-pack"
+BREW_IOS="swiftlint carthage"
+BREW_MACOS="carthage cocoapods"
+BREW_ANDROID="openjdk@8 android-sdk android-studio ant gradle maven"
+
+brew install $BREW_ALL
+$REQUIRE_ANDROID && brew install $BREW_ANDROID
+$REQUIRE_IOS && brew install $BREW_IOS
+$REQUIRE_WEB && brew install $BREW_WEB
+$REQUIRE_MACOS && brew install $BREW_MACOS
+
+$REQUIRE_PANDOC && brew install pandoc
+$REQUIRE_SENTRYCLI && brew install getsentry/tools/sentry-cli
+
+# Install WINE
+$REQUIRE_KMCOMP && (
+  brew tap homebrew/cask-versions
+  brew install --cask --no-quarantine wine-stable
+)
+
+$REQUIRE_WEB && (
+    rustup-init -y -t wasm32-unknown-unknown
+)
+
+source "$THIS_DIR/keyman.macos.env.sh"
+
+$REQUIRE_ANDROID && (
+    mkdir -p .android && touch ~/.android/repositories.cfg
+
+    which sdkmanager || (
+        echo "Please run Android Studio to install the Android SDK tools."
+        read -p "Press ENTER to continue after installation"
+        PATH="$HOME/Library/Android/sdk/tools/bin:$PATH"
+    )
+    sdkmanager "system-images;android-30;google_apis_playstore;x86_64"
+    sdkmanager --update
+    sdkmanager --licenses
+)
+
+($REQUIRE_IOS || $REQUIRE_MACOS) && (
+    # Assumes that xcode is installed into normal path
+    # as otherwise we get only the command line tools which
+    # won't build with xcode
+    sudo xcode-select -s /Applications/Xcode.app
+)
+
+# Add macos.env.sh to ~/.bashrc
+
+if [ ! -f ~/.bashrc ] || `grep "keyman.macos.env.sh" ~/.bashrc 2>/dev/null`; then
+    echo "source $THIS_DIR/keyman.macos.env.sh" >> ~/.bashrc
+fi
+
```

Comment:
One extra suggestion:  emit an echo message regarding https://github.com/keymanapp/keyman/wiki/How-to-setup-conventional-commits-git-commit-msg-hook.  Or add a parameter to auto-execute it, if you're feeling adventurous.

While not _necessary_ for development, that's highly suggested for contributors.

Though... I suppose we aren't auto-installing `git` for developers, so _maybe_ it falls out of scope?  Or maybe that'd best be added to a later pass - I can't help but note there's no base-level "prepare-dev-environment.sh".

Reply:
Yeah, I think that's a good idea but probably belongs in a CONTRIBUTORS.md. I'll put it into a separate doc.